### PR TITLE
Interpolate tile debug visualization during globe-Mercator transition

### PR DESCRIPTION
**Pull request by SnailBones**
Fixes https://github.com/mapbox/mapbox-gl-js/issues/11873 and https://github.com/mapbox/mapbox-gl-js/pull/12118#pullrequestreview-1054810759

The new function ` transitionTileAABBinECEF` allows for generating tile matrices that take the transition into account. This is important since using the regular tile matrices (generated by `globeTileBounds`) resulted in visual artifacts, likely because the extent of the actual tiles is larger than the area encompassed by the matrix.

I looked into updating `globeTileBounds` instead of adding a separate function (and additional matrix generation), unfortunately this is nontrivial since it is called by buckets, which don't currently have access to the transform's zoom and globe matrix.

Before:

image

After:

image

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [X] include before/after visuals or gifs if this PR includes visual changes
 - [ ] write tests for all new functionality
 - [ ] document any changes to public APIs
 - [ ] post benchmark scores
 - [ ] manually test the debug page
 - [ ] tagged `@mapbox/map-design-team` `@mapbox/static-apis` if this PR includes style spec API or visual changes
 - [ ] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog></changelog>`

### CODE REVIEW

File: src/source/tile.js [hunk 1]
```diff
@@ -769,7 +804,17 @@ class Tile {
...
+
+                // When in globe-to-Mercator transition, interpolate between globe and Mercator positions in ECEF
+                if (worldToECEFMatrix) {
+                    const tileCount = 1 << id.z;
+                    const mercatorX = (x / EXTENT + id.x) / tileCount + wrap;
+                    const mercatorY = (y / EXTENT + id.y) / tileCount;
+                    const mercatorPos = [mercatorX * tr.worldSize, mercatorY * tr.worldSize, 0];
+                    vec3.transformMat4(mercatorPos, mercatorPos, worldToECEFMatrix);
+                    ecef = interpolateArray(ecef, mercatorPos, phase);
+                }
```
Comment (karimnaaji) [inline src/source/tile.js:817]: Maybe share that with this [line](https://github.com/mapbox/mapbox-gl-js/pull/12118/files#diff-3dd84d28ccd48c2fc3addeef384f5ae733241d8236a6fcaa1e5fd7b0f2ce6835R753)?

**Review by karimnaaji — COMMENTED** [review]
Looks good to me, could you remove the extra commit that aren't from this branch? Also, I'm noticing an issue after the threshold globe - mercator, where sometimes tile bounds aren't rendered properly. I'm not sure if it's a regression incoming from this PR:

image

**Review by karimnaaji — APPROVED** [review]
Approving as it looks like the other issue also happens in `main`, could you please open an issue for follow-up if you don't address it in this PR?

**Comment by SnailBones** [comment]
> Approving as it looks like the other issue also happens in `main`, could you please open an issue for follow-up if you don't address it in this PR?

Just pushed a fix for this issue! The issue was that `tile._tileDebugIndexBuffer` was persisting during the projection change, by reseting it to `null` in `unloadVectorData` it correctly falls back to the Mercator version.

**Pull request merged**